index.js

```diff
@@ -1 +1,72 @@
-// implement your API here
+const express = require('express');
+const cors = require('cors');
+const db = require('./data/db');
+const server = express();
+
+server.use(express.json());
+server.use(cors());
+
+// C - POST
+server.post('/api/users', (req, res) => {
+  const { body: user } = req;
+
+  if (!user.name || !user.bio) {
+    return res.status(400).json({ errorMessage: 'Please provide name and bio for the user.' });
+  }
+
+  db.insert(user).then(
+    userDocument => res.status(201).json(userDocument),
+    error => res.status(500).json({ error: `There was an error while saving the user to the database; ${error}` })
```

Comment:
nice using the optional second optional argument of the `.then()` structure as a cleaner alternative to writing out the actual `.catch()` block!

Reply:
follow-up: I did a little more research on the pros and cons of this pattern, and while it is much cleaner to read and write, it also has a flaw which leads to being considered a bit of an antipattern--essentially the error callback will only trigger and send an error message if an erroneous promise gets returned by the `.insert` method, but not if an error occurs anywhere in your `success` callback of `userDocument => res.status(201).json(userDocument)`. While not necessarily an issue here, it will be if performing more complex functions on the response sent back with a success. The most relevant link describing the problem: https://stackoverflow.com/questions/24662289/when-is-thensuccess-fail-considered-an-antipattern-for-promises, check that diagram for a visualization of it.

That said, I'm still very impressed you're looking for ways to optimize any code you can :smile: